src/Main.elm

```diff
@@ -114,7 +112,17 @@ drop menu isMenuOpen =
         , dropdownMenu []
             []
             (List.range 1 10
-                |> List.map (\check -> dropdownItem False [] [ checkBox menu check ])
+                |> List.map
+                    (\num ->
+                        dropdownItem False
+                            []
+                            [ controlCheckBox False
+                                []
+                                [ onClickAndStop <| UpdateChoice menu num ]
+                                [ onClickAndStop <| NoOp ]
```

Comment:
I think you also should have explicit `checked` attribute here. Otherwise the fact that state of Dom and model is in sync (checked/unchecked) is coincidence.